lib/index.js

```diff
@@ -2237,6 +2237,53 @@ addMethod('flatMap', function (f) {
     return this.map(f).sequence();
 });
 
+/**
+ * Creates a new Stream of values by applying each item in a Stream to each
+ * value in a Stream of functions
+ *
+ * @id ap
+ * @section Higher-order Streams
+ * @name Stream.ap(m)
+ * @param {Stream} m - incoming stream of function(s) to apply to value(s) in stream
+ * @api public
+ *
+ * var asyncUnit = () => h.of(h.nil);
+ * var asyncAutocomplete = e => h(fetch(`/autocomplete?s=${e.target.value}`));
+ *
+ * var fns = h('change', checkbox)
+ *   .flatMap(e => h.of(e.target.checked ? asyncAutocomplete : asyncUnit));
+ *
+ * h('change', input)
+ *   .ap(fns)
+ *   .merge()
```

Comment:
Use `sequence` instead of `merge`. Strictly speaking, `merge` doesn't guaranteed an order, so it's not strictly correct.